x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -108,22 +128,87 @@ export const CreateField = React.memo(function CreateFieldComponent({
   const [nameValue, setNameValue] = useState<string>();
   const [inferenceIdComboValue, setInferenceIdComboValue] = useState<string>();
   const [semanticFieldType, setSemanticTextFieldType] = useState<string>();
+  const [showWarningAboutDeployment, setShowWarningAboutDeployment] = useState<boolean>(false);
```

Comment:
I think this file now has a lot of logic for the semantic text, it is possible to move it to a separate file? Also adding some unit tests would make it easier to maintain later

Reply:
Moved to a sperate file and also added additional unit tests.